### PR TITLE
Promote API metadata return value

### PR DESCRIPTION
**Pull request by BobyMCbobs**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Promotes the following E2E test to Conformance:
test/e2e/apimachinery/table_conversion.go: "should return a 406 for a backend which does not implement metadata"

**Special notes for your reviewer**:
Part of Umbrella Issue #78747.
[Testgrid results](https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-default&show-stale-tests=&include-filter-by-regex=which%20does%20not%20implement%20metadata&include-filter-by-regex=which%20does%20not%20implement%20metadata&include-filter-by-regex=return%20a%20406%20for%20a%20backend%20which%20does%20not%20implement%20metadata&include-filter-by-regex=should%20return%20a%20406%20for%20a%20backend%20which%20does%20not%20implement%20metadata&width=5).
Test(s):
- [sig-api-machinery] Servers with support for Table transformation should return a 406 for a backend which does not implement metadata

Endpoint(s):
- createAuthorizationV1SelfSubjectAccessReview

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```
/area conformance
/area test
@kubernetes/sig-apimachinery-pr-reviews
@kubernetes/sig-architecture-pr-reviews
@kubernetes/cncf-conformance-wg

### CODE REVIEW

File: test/e2e/apimachinery/table_conversion.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 	"github.com/onsi/gomega"
 
 	authorizationv1 "k8s.io/api/authorization/v1"
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (hh) [inline test/e2e/apimachinery/table_conversion.go]: Is gofmt adding v1?

Seems like it's trying to distinguish between the two v1s we are importing.

Comment (liggitt) [reply test/e2e/apimachinery/table_conversion.go]: > Is gofmt adding v1?

yes

File: test/e2e/apimachinery/table_conversion.go [hunk 2]
```diff
@@ -145,7 +145,13 @@ var _ = SIGDescribe("Servers with support for Table transformation", func() {
...
 	})
 
-	ginkgo.It("should return a 406 for a backend which does not implement metadata", func() {
+	/*
+			    Release : v1.16
+				Testname: API metadata HTTP return
+				Description: Issue a HTTP request to the API.
+		        HTTP request MUST return a HTTP status code of 406.
+	*/
+	framework.ConformanceIt("should return a 406 for a backend which does not implement metadata", func() {
```
Comment (liggitt) [inline test/e2e/apimachinery/table_conversion.go]: before promoting this to conformance, I'd expect it to request `v=v1`

Comment (dvn0) [reply test/e2e/apimachinery/table_conversion.go]: @liggitt do you think this is a small enough change that we could include it in this PR?

File: test/e2e/apimachinery/table_conversion.go (hunk 1 above)
Comment (dvn0) [reply test/e2e/apimachinery/table_conversion.go]: I've changed it back

File: test/e2e/apimachinery/table_conversion.go (hunk 2 above)
Comment (liggitt) [reply test/e2e/apimachinery/table_conversion.go]: yes

File: test/e2e/apimachinery/table_conversion.go (hunk 1 above)
Comment (devaips) [reply test/e2e/apimachinery/table_conversion.go]: I've changed it back.

File: test/e2e/apimachinery/table_conversion.go (hunk 2 above)
Comment (devaips) [reply test/e2e/apimachinery/table_conversion.go]: Deleted comment to re-draft, before I saw your response.

Original message:
> @liggitt do you think this is a small enough change that we could include it in this PR?

**Comment by liggitt** [comment]
ordering of conformance txt file is wrong.

This is a reasonable thing to conformance-test, though it applies to more resources than just this one. I'm not sure what the overall approach to systemic API testing is intended to be for conformance... spot check representative resources or do exhaustive testing of all of them?

**Comment by johnbelamaric** [comment]
I don't think exhaustive checks on different resources for the same core API server functionality is necessary. For example, for Watch conformance we can test one or two standard resources. As long as it's not resource-specific functionality that should be sufficient.

**Pull request merged**